bolero/optimizer/creps.py

```diff
@@ -53,9 +53,18 @@ def solve_dual_contextual_reps(S, R, epsilon, min_eta):
     def g(x):  # Objective function
         eta = x[0]
         nu = x[1:]
-        return (eta * epsilon + nu.T.dot(S.mean(axis=0)) +
-                eta * logsumexp((R - nu.dot(S.T)) / eta,
-                                b=1.0 / n_samples_per_update))
+
```

Comment:
I tried to improve the efficiency of the code but noticed that you did a pretty good job already. :)